integrationTests/mock/contracts/elrond.go

```diff
@@ -44,12 +52,71 @@ func (ec *ElrondContract) createContractFunctions() {
 	ec.AddHandler("getCurrentTxBatch", ec.getCurrentTxBatch)
 	ec.AddHandler("userRole", ec.userRole)
 	ec.AddHandler("getTokenIdForErc20Address", ec.getTokenIdForErc20Address)
+	ec.AddHandler("wasTransferActionProposed", ec.wasTransferActionProposed)
+	ec.AddHandler("proposeMultiTransferEsdtBatch", ec.proposeMultiTransferEsdtBatch)
+	ec.AddHandler("getActionSignerCount", ec.getActionSignerCount)
+	ec.AddHandler("proposeEsdtSafeSetCurrentTransactionBatchStatus", ec.proposeEsdtSafeSetCurrentTransactionBatchStatus)
 }
 
 func (ec *ElrondContract) getCurrentTxBatch(caller string, value string, arguments ...string) ([][]byte, error) {
 	log.Debug("getCurrentTxBatch", "caller", caller, "value", value, "arguments", arguments)
 
-	return make([][]byte, 0), nil
+	return make([][]byte, 0), nil //ec.currentTxBatch
+}
+
+func (ec *ElrondContract) wasTransferActionProposed(caller string, value string, arguments ...string) ([][]byte, error) {
+	log.Warn("wasTransferActionProposed", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+	batchId, _ := strconv.ParseInt(arguments[0], 10, 64)
+	ret := byte(1) //we assume we have the action in map
+	if _, ok := ec.action_mapper[big.NewInt(batchId)]; !ok {
+		ret = 0
+	}
+	return [][]byte{{ret}}, nil
+}
+func (ec *ElrondContract) proposeMultiTransferEsdtBatch(caller string, value string, arguments ...string) ([][]byte, error) {
+	log.Warn("proposeMultiTransferEsdtBatch ", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+	var args []string
+	err := json.Unmarshal([]byte(arguments[0]), &args)
+	if err != nil {
+		log.Error("ElrondContract: Error unmarshal arguments", "error", err.Error())
+		return make([][]byte, 0), nil
+	}
+	buf, _ := base64.URLEncoding.DecodeString(args[0])
+	batchId := int64(buf[0])
```

Comment:
shouldn't we convert otherwise this value? What happens if the buff is 2 characters long?